addons/arsenal/functions/fnc_fillLeftPanel.sqf

```diff
@@ -113,6 +113,21 @@ private _selectedItem = if (_idxVirt != -1) then { // Items
                 ["CfgUnitInsignia", _x, _ctrlPanel, "texture", _y] call FUNC(addListBoxItem);
             } forEach GVAR(insigniaCache);
 
+            // If the current insignia is not in the arsenal, add it manually
+            if (GVAR(currentInsignia) != "" && {!(GVAR(currentInsignia) in GVAR(insigniaCache))}) then {
+                // Check for correct config: First mission, then campaign and finally regular config
+                // Ref fnc_addListBoxItem, 0/nil = configFile, 1 = campaignConfigFile, 2 = missionConfigFile
+                private _index = 3;
+
+                {
+                    _index = _index - 1;
+
+                    if (!isNull (_x >> "CfgUnitInsignia" >> GVAR(currentInsignia))) exitWith {};
+                } forEach [missionConfigFile, campaignConfigFile, configFile];
+
+                ["CfgUnitInsignia", GVAR(currentInsignia), _ctrlPanel, "texture", _index] call FUNC(addListBoxItem);
+            };
+
```

Comment:
```suggestion
```

Discussed in DMs but TL;DR is for this condition to be met then the mission maker screwed up or the user has scripting access

IMO, unnecessary

Reply:
It could be in one of their saved loadouts with CBA extended loadouts, unless that has a check to make sure it is in the arsenal, I think we do for equipment, not sure about the insignia